### PR TITLE
[FEATURE] NGSI v1 vs NGSI v2 comparison

### PR DESCRIPTION
**Pull request by iariasleon**

### CODE REVIEW

File: test/loadTest/v1_vs_v2.sh [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+  fi
+  shift
+done
+
+if [ "$SUBSC" == "true" ]; then
+   curl "http://$HOST:$PORT/v2/subscriptions" -H "Fiware-service: test" -H "Fiware-ServicePath: /" -H "Content-Type: application/json" -d '{"subject": {"entities": [{"type": "Room","idPattern": ".*"}], "condition": {"attrs": ["temperature"]}}, "notification": {"http": {"url": "'$SUBSC_REFERENCE'"}, "attrs": ["temperature"]}}' &> /dev/null
+   echo "...a subscription has been created..."
+else
+    echo "...Executing "$UPDATE_TOTAL "updates using NGSI "$NGSI_VERSION "in the host "$HOST "..."
+    for (( c=1; c<=$UPDATE_TOTAL; c++ ))
```
Comment (fgalan) [inline test/loadTest/v1_vs_v2.sh]: I'd suggest to move `INIT_DATE=$(date +%s)` just before the `for`, to be more accuarte measuinig the time interval.

File: test/loadTest/v1_vs_v2.sh [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+           VALUE=$(random_string 8)
+           if [ "$NGSI_VERSION" == "v1" ]; then
+              curl  "http://$HOST:$PORT/v1/updateContext" -H "Fiware-service: test" -H "Fiware-ServicePath: /" -H "Content-Type: application/json" -d '{"contextElements": [{"type": "Room", "isPattern": "false", "id": " Room1", "attributes": [{"name": "temperature", "value": "'$VALUE'"}]}], "updateAction": "APPEND"}' &> /dev/null
+           else
+              curl "http://$HOST:$PORT/v2/op/update" -H "Fiware-service: test" -H "Fiware-ServicePath: /" -H "Content-Type: application/json" -d '{"entities": [{"type": "Room", "id": "Room1", "temperature": {"value": "'$VALUE'"}}], "actionType": "APPEND"}' &> /dev/null
+           fi
+        done
+
+    END_DATE=$(date + %s)
+    echo "the test using NGSI "$NGSI_VERSION " has delayed" $(( END_DATE - INIT_DATE )) "secs..."
```
Comment (fgalan) [inline test/loadTest/v1_vs_v2.sh]: has delayed -> has taken

File: test/loadTest/v1_vs_v2.sh [hunk 3]
```diff
@@ -0,0 +1,93 @@
+﻿#!/bin/bash
```
Comment (fgalan) [inline test/loadTest/v1_vs_v2.sh]: Rather than v1_vs_v2.sh, I'd suggest: update_time_measure.sh or something like that.

Comment (iariasleon) [inline test/loadTest/v1_vs_v2.sh]: Fixed in 5c22c38

File: test/loadTest/v1_vs_v2.sh [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+# -*- coding: latin-1 -*-
+# Copyright 2014 Telefonica Investigacion y Desarrollo, S.A.U
```
Comment (iariasleon) [inline test/loadTest/v1_vs_v2.sh]: Fixed in 5c22c38

File: test/loadTest/v1_vs_v2.sh (hunk 1 above)
Comment (iariasleon) [inline test/loadTest/v1_vs_v2.sh]: Fixed in 5c22c38

File: test/loadTest/v1_vs_v2.sh (hunk 2 above)
Comment (iariasleon) [inline test/loadTest/v1_vs_v2.sh]: Fixed in 5c22c38

File: test/loadTest/updates_time_measure.sh.sh [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+      echo $0: bad parameter/option: "'"${1}"'";
+      usage
+      exit
+  fi
+  shift
+done
+
+if [ "$SUBSC" == "true" ]; then
+   curl "http://$HOST:$PORT/v2/subscriptions" -H "Fiware-service: test" -H "Fiware-ServicePath: /" -H "Content-Type: application/json" -d '{"subject": {"entities": [{"type": "Room","idPattern": ".*"}], "condition": {"attrs": ["temperature"]}}, "notification": {"http": {"url": "'$SUBSC_REFERENCE'"}, "attrs": ["temperature"]}}' &> /dev/null
+   echo "...a subscription has been created..."
```
Comment (kzangeli) [inline test/loadTest/updates_time_measure.sh.sh]: If you want to do echo without newline, use `-n`
NTC

File: test/loadTest/updates_time_measure.sh.sh [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+#
+# Author: Ivan Arias Leon
+
+# variables
+HOST="localhost"
+PORT="1026"
+NGSI_VERSION="v1"
+UPDATE_TOTAL=100
+SUBSC=false
+SUBSC_REFERENCE="http://localhost:1028/notify"
```
Comment (kzangeli) [inline test/loadTest/updates_time_measure.sh.sh]: In shell, everything is treated as strings (well, more or less), so citation marks aren't necessary

    HOST=localhost

is valid too

NTC

File: test/loadTest/updates_time_measure.sh.sh [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+     echo "   --subsc_reference <listener url>"
+     echo " "
+     echo " Example:"
+     echo "     "$(basename $0)" -host localhost --version v2 --updates 10000 --subsc_reference http://localhost:1028/notify"
+     echo " "
+     echo " Note: "
+     echo "     - if the subscription is created, the entities are not updated and viceversa..."
+     echo "     - if \"--subsc_reference\" parameter is used is not necessary the \"-subsc\" parameter ..."
+     echo " "
+}
```
Comment (kzangeli) [inline test/loadTest/updates_time_measure.sh.sh]: You might want an exit 0 at the end of the usage function

File: test/loadTest/updates_time_measure.sh.sh [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+HOST="localhost"
+PORT="1026"
+NGSI_VERSION="v1"
+UPDATE_TOTAL=100
+SUBSC=false
+SUBSC_REFERENCE="http://localhost:1028/notify"
+
+function random_string(){
+    # a random string is generated with a given size
+    cat /dev/urandom | tr -dc 'a-zA-Z0-9' | fold -w ${1:-32} | head -n 1
```
Comment (kzangeli) [inline test/loadTest/updates_time_measure.sh.sh]: A bit cryptic use of $1

I usually receive the parameters and give them a name before I continue the function.
E.g.:

    # ========================================================
    # 
    # random_string - generate a random string with a given size
    # 
    function random_string()
    {
      size=$1
      cat /dev/urandom | tr -dc 'a-zA-Z0-9' | fold -w ${size:-32} | head -n 1
    }

Something like that. The function gets easier to understand.
Also, the function header makes it easier to browse the code.
You clearly see where a function starts and ends.

NTC

Comment (iariasleon) [inline test/loadTest/updates_time_measure.sh.sh]: Fixed in  	a36b421

File: test/loadTest/updates_time_measure.sh.sh (hunk 7 above)
Comment (iariasleon) [inline test/loadTest/updates_time_measure.sh.sh]: Fixed in  	a36b421

File: test/loadTest/updates_time_measure.sh.sh (hunk 6 above)
Comment (iariasleon) [inline test/loadTest/updates_time_measure.sh.sh]: Fixed in  	a36b421

**Comment by kzangeli** [comment]
LGTM

**Comment by fgalan** [comment]
LGTM

**Pull request merged**